mmtbx/domains_from_pae.py

```diff
@@ -38,29 +38,62 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 """
-def parse_pae_file(pae_json_file):
-    import json, numpy
 
-    try:
-      with open(pae_json_file, 'rt') as f:
-        data = json.load(f)[0]
-    except Exception as e:
-      raise Sorry("Unable to read the json file %s" %(pae_json_file))
 
-    r1, d = data['residue1'],data['distance']
+def parse_pae_file(pae_file):
+    """Takes PAE file in PKL or JSON format and returns PAE matrix
 
-    size = max(r1)
+    Arguments:
+        *pae_file: file containing PAE matrix in PKL or JSON format
 
-    matrix = numpy.empty((size,size))
+    Returns:
+        pae_matrix as numpy array
+    """
+    from pathlib import Path
+    import numpy
 
-    matrix.ravel()[:] = d
+    ext = Path(pae_file).suffix
+    if ext == '.json':
+        import json
+
+        try:
+            with open(pae_file, 'rt') as f:
+                data = json.load(f)[0]
```

Comment:
To clarify, the ColabFold PAE files will fail with an `IndexError` here since they only contain a dict, whereas an AlphaFold PAE file contains a list with the dict as the first entry.

Reply:
I've fixed this and tested it against every PAE variant so ti should all be working now :)

Reply:
Looks good!

Reply:
Thanks very much Adam and Tristan!